### DISCUSSION TITLE
Error on everything but windows - System.MissingMethodException: Method not found: void Microsoft.UI.Xaml.Controls.Grid.SetColumn(Microsoft.UI.Xaml.UIElement,int)

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno#16185] [Q&A]
### Current behavior

I have a fairly basic xaml file, it works fine on winui3 but when i try it in Maccatalyst, iOS or android i get the following error

```
Microsoft.UI.Xaml.UIElement: Error: Failed to raise 'PointerReleasedEvent': System.InvalidOperationException: Failed to load ZiiDMSPhoneApp.PartsSearch: System.MissingMethodException: Method not found: void Microsoft.UI.Xaml.Controls.Grid.SetColumn(Microsoft.UI.Xaml.UIElement,int)
   at Uno.UI.Helpers.Xaml.ApplyExtensions.GenericApply[ZtLabel,PartsSearch,NameScope](ZtLabel instance, PartsSearch arg1, NameScope arg2, Action`3 apply)
   at ZiiDMSPhoneApp.PartsSearch.InitializeComponent() in /Users/curtisconner/ZiiDMSV2/Client/Mobile/ZiiDMSPhoneApp/ZiiDMSPhoneApp/Uno.UI.SourceGenerators/Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator/PartsSearch_3cb178e422e1aa7286746267d426acc3.cs:line 77
   at ZiiDMSPhoneApp.PartsSearch..ctor() in /Users/curtisconner/ZiiDMSV2/Client/Mobile/ZiiDMSPhoneApp/ZiiDMSPhoneApp/Views/Parts/PartsSearch.xaml.cs:line 12
   at ZiiDMSPhoneApp.Mobile.MetadataBuilder_722.CreateInstance() in /Users/curtisconner/ZiiDMSV2/Client/Mobile/ZiiDMSPhoneApp/ZiiDMSPhoneApp.Mobile/Uno.UI.SourceGenerators/Uno.UI.SourceGenerators.BindableTypeProviders.BindableTypeProvidersSourceGenerator/BindableMetadata.g.cs:line 29382
   at Microsoft.UI.Xaml.Controls.Frame.CreatePageInstance(Type sourcePageType)
   at Microsoft.UI.Xaml.PagePool.DequeuePage(Type pageType)
   at Microsoft.UI.Xaml.Controls.Frame.CreatePageInstanceCached(Type sourcePageType)
   at Microsoft.UI.Xaml.Controls.Frame.EnsurePageInitialized(PageStackEntry entry)
   at Microsoft.UI.Xaml.Controls.Frame.InnerNavigate(PageStackEntry entry, NavigationMode mode)
   at ZiiDMSPhoneApp.App.OnNavigationFailed(Object sender, NavigationFailedEventArgs e) in /Users/curtisconner/ZiiDMSV2/Client/Mobile/ZiiDMSPhoneApp/ZiiDMSPhoneApp/App.cs:line 258
   at Microsoft.UI.Xaml.Controls.Frame.InnerNavigate(PageStackEntry entry, NavigationMode mode)
   at Microsoft.UI.Xaml.Controls.Frame.Navigate(Type sourcePageType, Object parameter, NavigationTransitionInfo infoOverride)
   at Microsoft.UI.Xaml.Controls.Frame.Navigate(Type sourcePageType)
   at ZiiDMSPhoneApp.MainMenuViewModel.NavigateToPage(Type page) in /Users/curtisconner/ZiiDMSV2/Client/Mobile/ZiiDMSPhoneApp/ZiiDMSPhoneApp/MainMenuViewModel.cs:line 59
   at ZiiDMSPhoneApp.MainMenuViewModel.Parts() in /Users/curtisconner/ZiiDMSV2/Client/Mobile/ZiiDMSPhoneApp/ZiiDMSPhoneApp/MainMenuViewModel.cs:line 29
   at CommunityToolkit.Mvvm.Input.RelayCommand.Execute(Object parameter)
   at Microsoft.UI.Xaml.Controls.Primitives.ButtonBase.ExecuteCommand()
   at Microsoft.UI.Xaml.Controls.Primitives.ButtonBase.OnClick()
   at Microsoft.UI.Xaml.Controls.Button.OnClick()
   at Microsoft.UI.Xaml.Controls.Primitives.ButtonBase.PerformPointerUpAction()
   at Microsoft.UI.Xaml.Controls.Primitives.ButtonBase.OnPointerReleased(PointerRoutedEventArgs args)
   at Microsoft.UI.Xaml.Controls.Control.<>c.<.cctor>b__366_10(Object sender, PointerRoutedEventArgs args)
   at Microsoft.UI.Xaml.UIElement.InvokeHandler(Object handler, RoutedEventArgs args)
   at Microsoft.UI.Xaml.UIElement.RaiseEvent(RoutedEvent routedEvent, RoutedEventArgs args, BubblingContext ctx)
   at Microsoft.UI.Xaml.UIElement.RaisePointerEvent(RoutedEvent evt, PointerRoutedEventArgs args, BubblingContext ctx)
```

the xaml file is simple

```
<Page
  xmlns:utu="using:Uno.Toolkit.UI"
  xmlns:muxc="using:Microsoft.UI.Xaml.Controls"
  xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
  xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
  xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
  xmlns:local="using:ZiiDMSPhoneApp"
  xmlns:ztcf="using:ZtClientFramework"
  x:Class="ZiiDMSPhoneApp.PartsSearch"
  mc:Ignorable="d"
  Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
  xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">

  <Page.Resources>
    <x:String x:Key="Icon_Arrow_Back">F1 M 16 7 L 3.8299999237060547 7 L 9.420000076293945 1.4099998474121094 L 8 0 L 0 8 L 8 16 L 9.40999984741211 14.59000015258789 L 3.8299999237060547 9 L 16 9 L 16 7 Z</x:String>
  </Page.Resources>

  <StackPanel>

    <utu:NavigationBar HorizontalContentAlignment="Left" Content="Parts Inventory" utu:AutoLayout.CounterAlignment="Stretch">
      <utu:NavigationBar.MainCommand>
        <AppBarButton>
          <AppBarButton.Icon>
            <PathIcon Data="{StaticResource Icon_Arrow_Back}" />
          </AppBarButton.Icon>
        </AppBarButton>
      </utu:NavigationBar.MainCommand>
    </utu:NavigationBar>

    <ztcf:ZtLabel HorizontalAlignment="Center" TextWrapping="Wrap" Text="{x:Bind ViewModel.DealershipName, Mode=OneWay}" utu:AutoLayout.PrimaryAlignment="Stretch"  />
    
    <ztcf:ZtAutoSuggestBox TextChanged="ZtAutoSuggestBox_TextChanged"
                           WaterMark="Enter part no or number"
                           QuerySubmitted="ZtAutoSuggestBox_QuerySubmitted"
                           CornerRadius="12"                           
                           HorizontalAlignment="Stretch"/>
    <Grid>
      <Grid.ColumnDefinitions>
        <ColumnDefinition Width="*"/>
        <ColumnDefinition Width="Auto"/>
      </Grid.ColumnDefinitions>
      <Grid.RowDefinitions>
        <RowDefinition Height="Auto"/>
      </Grid.RowDefinitions>
      <ztcf:ZtLabel Text="{x:Bind ViewModel.ResultCount, Mode=OneWay}" HorizontalAlignment="Left" Grid.Column="0" Grid.Row="0" utu:AutoLayout.CounterAlignment="Stretch" />
      <HyperlinkButton Content="Filters" Grid.Column="1" Grid.Row="0" HorizontalAlignment="Right" Command="{x:Bind ViewModel.FiltersCommand}"/>
    </Grid>


    <utu:Divider utu:AutoLayout.CounterAlignment="Stretch" />
    <ListView ItemsSource="{x:Bind ViewModel.SearchResults}" ItemClick="ListView_ItemClick"  IsItemClickEnabled="True" SelectionMode="Single">
      <ListView.ItemTemplate>
        <DataTemplate x:Name="PartsSearchTemplate" x:DataType="local:SearchResultTemplate">
          <Grid>
            <Grid.ColumnDefinitions>
              <ColumnDefinition Width="Auto"/>
              <ColumnDefinition Width="*"/>
            </Grid.ColumnDefinitions>
            <Grid.RowDefinitions>
              <RowDefinition Height="Auto"/>
              <RowDefinition Height="Auto"/>
              <RowDefinition Height="Auto" />
            </Grid.RowDefinitions>
            <ztcf:ZtLabel Text="{x:Bind PartNo}" Grid.Column="0" Grid.Row="0" TextAlignment="Left" FontSize="10" FontStyle="Normal" utu:AutoLayout.CounterAlignment="Stretch" />
            <ztcf:ZtLabel Text="{x:Bind Vendor}" Grid.Column="1" Grid.Row="0" TextAlignment="Right" FontSize="10" FontStyle="Normal" utu:AutoLayout.CounterAlignment="Stretch" />

            <ztcf:ZtLabel Text="{x:Bind Description}" Grid.Column="0" Grid.Row="1" TextAlignment="Left" FontSize="14" FontStyle="Italic" utu:AutoLayout.CounterAlignment="Stretch" />
            <ztcf:ZtLabel Text="{x:Bind QtyAvailable}" Grid.Column="1" Grid.Row="1" TextAlignment="Right" FontSize="14" FontStyle="Italic" utu:AutoLayout.CounterAlignment="Stretch" />

            <ztcf:ZtLabel Text="{x:Bind Location}" Grid.Column="0" Grid.Row="2" TextAlignment="Left" FontSize="10" FontStyle="Normal" utu:AutoLayout.CounterAlignment="Stretch" />
            <ztcf:ZtLabel Text="{x:Bind Price}" Grid.Column="1" Grid.Row="2" TextAlignment="Right" FontSize="10" FontStyle="Normal" utu:AutoLayout.CounterAlignment="Stretch" />
          </Grid>
        </DataTemplate>
      </ListView.ItemTemplate>
    </ListView>
  </StackPanel>
</Page>
```

Is there anything obvious i am doing wrong?

### Expected behavior

Should work the same on other platforms as it does on windows Winui3

### Works on UWP/WinUI

Yes

**Comment by Youssef1313** [comment]
Thanks for the report @cconner100 

Can you attach the project as a zip file? Thank you!

**Reply by cconner100** [reply]
is it possible to give you access to my github repository? that would be simpler, what's the id to allow access to

**Comment by jeromelaban** [comment]
Also, in general, this means that there's an Uno 4.x reference somewhere.

**Reply by cconner100** [reply]
OK here is the list of nugets i am referencing
```
<PackageVersion Include="Uno.WinUI.MSAL" Version="5.1.104" />
<PackageVersion Include="Uno.WinUI.Markup" Version="5.0.13" />
<PackageVersion Include="Uno.Extensions.Markup" Version="5.0.13" />
<PackageVersion Include="Uno.Extensions.Markup.Generators" Version="5.0.13" />
<PackageVersion Include="Uno.Extensions.Navigation.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Navigation.Toolkit.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Toolkit.WinUI" Version="5.1.7" />
<PackageVersion Include="Uno.Toolkit.WinUI.Material" Version="5.1.7" />
<PackageVersion Include="Uno.Toolkit.WinUI.Markup" Version="5.0.19" />
<PackageVersion Include="Uno.Material.WinUI" Version="4.1.1" />
<PackageVersion Include="Uno.Material.WinUI.Markup" Version="4.0.6" />
<PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
<PackageVersion Include="Uno.Extensions.Core.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Configuration" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Hosting" Version="3.0.11" />
<PackageVersion Include="Uno.Extensions.Hosting.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Localization" Version="3.0.11" />
<PackageVersion Include="Uno.Extensions.Localization.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Extensions.Logging.OSLog" Version="1.7.0" />
<PackageVersion Include="Uno.Extensions.Logging.WebAssembly.Console" Version="1.7.0" />
<PackageVersion Include="Uno.Extensions.Logging.WinUI" Version="4.0.0" />
<PackageVersion Include="Uno.Resizetizer" Version="1.3.0" />
<PackageVersion Include="Uno.Dsp.Tasks" Version="1.2.9" />
<PackageVersion Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="5.1.104" />
<PackageVersion Include="Uno.UniversalImageLoader" Version="1.9.36" />
<PackageVersion Include="Uno.Wasm.Bootstrap" Version="8.0.11" />
<PackageVersion Include="Uno.Wasm.Bootstrap.DevServer" Version="8.0.11" />
<PackageVersion Include="Uno.Wasm.Bootstrap.Server" Version="8.0.11" />
<PackageVersion Include="Uno.WinUI" Version="5.1.104" />
<PackageVersion Include="Uno.WinUI.Lottie" Version="5.1.104" />
<PackageVersion Include="Uno.WinUI.DevServer" Version="5.1.104" />
<PackageVersion Include="Uno.WinUI.WebAssembly" Version="5.1.87" />
<PackageVersion Include="Uno.UITest.Helpers" Version="1.1.0-dev.70" />
```
I do notice many 4.0 items but when i check nuget.org this is the latest version of these, am i referencing the wrong package names?

**Reply by jeromelaban** [reply]
Thanks. It's indeed difficult to determine which is which in a list like this, and this is something we're fixing in our next release.

That being said, it's possible to find out what the assembly that contains `ZiiDMSPhoneApp.PartsSearch` is referencing Uno 4.x somehow. A [binlog could help determining](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) why this happens.

**Reply by Youssef1313** [reply]
@jeromelaban Another possible cause is somehow have mixed references of `net7.0` and `net7.0-android` (but not sure if that's really the case or how this could happen).

**Reply by cconner100** [reply]
Here is the binlog compressed it was built on windows, do you need one thats built on ios?
[msbuild.zip](https://github.com/unoplatform/uno/files/14939983/msbuild.zip)

**Reply by jeromelaban** [reply]
Thanks. Looking at the binlog, I can see that there are still references to the community toolkit 8.0, which is referencing Uno 4.x:

![image](https://github.com/unoplatform/uno/assets/5839577/771bfed1-f0fb-4983-a74d-c75fe5e70921)

Try updating those packages to the 8.1 preview release and see if it helps.

**Reply by cconner100** [reply]
thanks,

I updated the community toolkit to the rc1 but now get this error in android

C:\Users\curtisconner\source\repos\ZiiDMSV2\Framework\Client\ZtClientFramework\ZtClientFramework\Controls\ZtMaskEdit\ZtMaskEdi
t.xaml.cs(164,17): error CS0012: The type 'TextBox' is defined in an assembly that is not referenced. You must add a reference
 to assembly 'Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null'. [C:\Users\curtisconner\source\repos\ZiiD
MSV2\Framework\Client\ZtClientFramework\ZtClientFramework\ZtClientFramework.csproj::TargetFramework=net8.0-windows10.0.19041]
C:\Users\curtisconner\source\repos\ZiiDMSV2\Framework\Client\ZtClientFramework\ZtClientFramework\Controls\ZtComboBox\ZtComboBo
x.xaml.cs(310,25): error CS0012: The type 'FrameworkElement' is defined in an assembly that is not referenced. You must add a
reference to assembly 'Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null'. [C:\Users\curtisconner\source\r
epos\ZiiDMSV2\Framework\Client\ZtClientFramework\ZtClientFramework\ZtClientFramework.csproj::TargetFramework=net8.0-windows10.
0.19041]

Seems like the TextBoxExtensions are not correct for android yet

**Reply by jeromelaban** [reply]
Thanks. The error you're getting is for windows (see the end of each error line), and it's caused by the update to 8.1, which requires an updated version of the Windows SDK. You'll need to change all `net8.0-windows10.0.19041` to `net8.0-windows10.0.22621`, as mentioned in https://github.com/CommunityToolkit/Windows/issues/388#issuecomment-2043166429. The error will change in an updated version of the toolkit to be a bit less cryptic.

**Reply by cconner100** [reply]
ok got the windows version to compile and run, but on MacCatalyst, iOS and Android i still get the previous error, here is the new binlog

[Uploading msbuild.zip…]()

**Reply by cconner100** [reply]
reuploading file

[msbuild.zip](https://github.com/unoplatform/uno/files/14940994/msbuild.zip)

**Reply by Youssef1313** [reply]
@cconner100 I can see why this may be happening on Catalyst, but not yet seeing it for Android/iOS.

For Catalyst, you have `ZiiDMSPhoneApp.Mobile.csproj` which targets `net8.0-android;net8.0-ios;net8.0-maccatalyst`. This project is referencing `ZiiDMSPhoneApp.csproj` which targets `net8.0-windows10.0.22621;net8.0;net8.0-android;net8.0-ios`. So, the net8.0-maccatalyst artifacts of the Mobile project will be referencing the `net8.0` artifacts of `ZiiDMSPhoneApp` (which is compiled against Uno's reference API which in this case has differences in the public API compared with `net8.0-maccatalyst`

In short, `ZiiDMSPhoneApp.csproj` needs to target `maccatalyst` as well.

Are you able to re-confirm that the error is still happening on Android/iOS?

**Reply by cconner100** [reply]
Thank you ever so much for this, it now works when i changed to include the maccatalyst.

thanks again for the help

**Reply by jeromelaban** [reply]
Thanks for the update. It looks like we should be having a validation that prevents that scenario from happening. We have similar checks for WinAppSDK. Let's track in https://github.com/unoplatform/uno/issues/16230.
